Common/cpp/ReanimatedRuntime/WorkletRuntime.cpp

```diff
@@ -4,15 +4,64 @@
 #include "WorkletRuntimeCollector.h"
 #include "WorkletRuntimeDecorator.h"
 
+#include <jsi/decorator.h>
+
 namespace reanimated {
 
+struct AroundLock {
+  std::recursive_mutex *mutex;
+  void before() {
+    mutex->lock();
+  }
+  void after() {
+    mutex->unlock();
+  }
+};
+
+class LockableRuntime : public jsi::WithRuntimeDecorator<AroundLock> {
+ private:
+  AroundLock aroundLock_;
+  std::shared_ptr<jsi::Runtime> runtime_;
+
+ public:
+  explicit LockableRuntime(
+      std::shared_ptr<jsi::Runtime> &&runtime,
+      std::recursive_mutex *runtimeMutex)
+      : jsi::WithRuntimeDecorator<AroundLock>(*runtime, aroundLock_),
+        runtime_(std::move(runtime)) {
+    aroundLock_.mutex = runtimeMutex;
+  }
```

Comment:
Does it work this way?
```suggestion
        runtime_(std::move(runtime)), aroundLock_(runtimeMutex) {}
```

Reply:
yup, but I need to add constructor to AroundLock